scripts/gen_datafile.py

```diff
@@ -0,0 +1,134 @@
+import math, random, string, os, time
+from dataclasses import dataclass
+from enum import auto, StrEnum
+from multiprocessing import Pool
+from typing import Iterator
+
+
+class ConfigNames(StrEnum):
+    KILO = auto()
+    PERCENT = auto()
+    FULL = auto()
+
+
+@dataclass
+class TestFileConfig:
+    total_rows: int
+    unique_locations: int
+    range: tuple[float, float]
+    configName: ConfigNames
+
+
+TEMP_RANGE = (-99.9, 99.9)
+
+CONFIGS: dict[ConfigNames, TestFileConfig] = {
+    ConfigNames.KILO: TestFileConfig(1_000, 50, TEMP_RANGE, ConfigNames.KILO),
+    ConfigNames.PERCENT: TestFileConfig(
+        10_000_000, 10_00, TEMP_RANGE, ConfigNames.PERCENT
+    ),
+    ConfigNames.FULL: TestFileConfig(
+        1_000_000_000, 10_000, TEMP_RANGE, ConfigNames.FULL
+    ),
+}
+
+
+class TestFileGen:
+    config: TestFileConfig
+    keys: list[str] = []
+    rows: list[str] = []
+    pageSize: int = 16_000
+    directory = "./.test_resources"
+    filenamePrefix = "test_data"
+    rowsProcessed = 0
+    iterations: Iterator[int]
+
+    def __init__(self, config: TestFileConfig) -> None:
+        t0 = time.time()
+        self.config = config
+        self.generateKeys()
+        self.cleanFiles()
+        self.iterations = iter(
+            range(math.ceil(self.config.total_rows / self.pageSize) - 1)
+        )
+
+        with Pool() as p:
+            self.rowsProcessed = sum(p.map(self.runIteration, self.iterations))
+
+        self.rowsProcessed += self.runIteration(
+            math.ceil(self.config.total_rows / self.pageSize), True
+        )
```

Comment:
Separating out the last iteration and passing in a bool as to whether the run was the last iteration, was a simple solution to preventing an empty row at the end of the file